### DISCUSSION TITLE
Selecting elements using Accessible Names

### DISCUSSION THREAD

**Discussion by crunchy-vonage** [cypress-io/cypress#18689] [General]
Hi All,

I'm interested in using accessibility names to find elements instead of CSS Selectors.

The accessible names of objects are very important for disabled access (screen readers), they tend to better looked after than automation IDs and have clever defaults if absent. There is also the social benefit of encouraging good accessibility of sites motivated by use of test automation.

Puppeteer folks have added support via devtools protocol:-
- https://developer.chrome.com/blog/puppetaria/

But there is a big push on browser accessibility recently (following webcomponents braking the existing model) and w3c are pushing a number of new API's to set the accessibility properties and access the result.
https://wicg.github.io/aom/caniuse.html

Support is coming to all major browsers but I have been playing with chrome with the following flag"--enable-blink-features=AccessibilityObjectModel".
This exposes **_window.getComputedAccessibleNode(el)_** which can be used to get the elements accessibly role and name.

Questions:-
* Did I convince you, is this of wide interest?
* Is there any prior discussion of cypress support?
* Are you interested in working with me to add support in cypress?
* Is there a simple way I could be a POC? - I couldn't workout the code,  does this sound like something easy?
Thanks Matt

**Comment by 787681277** [comment]
I also want to have this feature
